### ISSUE TITLE
App always tries to upsync attachments which are no longer available locally

### ISSUE THREAD

**Issue by neerro** [stefan-niedermann/nextcloud-deck#1075]
**Describe the bug**
Android app does not synch with server. Server is online and I have full access to it. 

1. Open app and see error message 
2. The same error when I tried synch manually.

**Versions**

 - Nextcloud: 20.0.1
 - Nextcloud Deck: 1.4.3
 - Nextcloud Android: 3.16.1
 - Nextcloud Android Deck: 1.19.3

**Smartphone (please complete the following information):**
- Device: Honor 10 light
- Android-Version: 10
- App-Store:
  - Google Play Store

**Stacktrace**

```
App Version: 1.19.3
App Version Code: 1019003
Server App Version: 1.4.3
App Flavor: play

Files App Version Code: 30160190

---

OS Version: 4.14.116(10.0.0.221C10)
OS API Level: 29
Device: HWHRY-H
Manufacturer: HUAWEI
Model (and Product): HRY-LX1 (HRY-LX1RU)

---

java.lang.IllegalStateException: failed to build request-body
	at com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod.bodyToStream(NextcloudRetrofitServiceMethod.java:225)
	at com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod.invoke(NextcloudRetrofitServiceMethod.java:164)
	at retrofit2.NextcloudRetrofitApiBuilder.lambda$create$0$NextcloudRetrofitApiBuilder(NextcloudRetrofitApiBuilder.java:30)
	at retrofit2.NextcloudRetrofitApiBuilder$$ExternalSyntheticLambda0.invoke(Unknown Source:2)
	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
	at $Proxy1.uploadAttachment(Unknown Source)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.lambda$uploadAttachment$30$ServerAdapter(ServerAdapter.java:293)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter$$ExternalSyntheticLambda27.getObservableFromCall(Unknown Source:12)
	at it.niedermann.nextcloud.deck.api.RequestHelper.request(RequestHelper.java:39)
	at it.niedermann.nextcloud.deck.persistence.sync.adapters.ServerAdapter.uploadAttachment(ServerAdapter.java:293)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.AttachmentDataProvider.createOnServer(AttachmentDataProvider.java:71)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.AttachmentDataProvider.createOnServer(AttachmentDataProvider.java:20)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper.doUpSyncFor(SyncHelper.java:125)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper.doUpSyncFor(SyncHelper.java:108)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider.goDeeperForUpSync(CardDataProvider.java:334)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.StackDataProvider.goDeeperForUpSync(StackDataProvider.java:129)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.BoardDataProvider.goDeeperForUpSync(BoardDataProvider.java:179)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper.doUpSyncFor(SyncHelper.java:129)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper.doUpSyncFor(SyncHelper.java:108)
	at it.niedermann.nextcloud.deck.persistence.sync.SyncManager$2$1.lambda$onResponse$0(SyncManager.java:237)
	at it.niedermann.nextcloud.deck.persistence.sync.SyncManager$2$1$$ExternalSyntheticLambda0.run(Unknown Source:4)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:929)
Caused by: java.io.FileNotFoundException: /data/user/0/it.niedermann.nextcloud.deck.play/cache/compressor/2021-07-28_BackGroundImage.jpg: open failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.open(IoBridge.java:496)
	at java.io.FileInputStream.<init>(FileInputStream.java:159)
	at okio.Okio.source(Okio.java:168)
	at okhttp3.RequestBody$3.writeTo(RequestBody.java:119)
	at okhttp3.MultipartBody.writeOrCountBytes(MultipartBody.java:173)
	at okhttp3.MultipartBody.writeTo(MultipartBody.java:114)
	at com.nextcloud.android.sso.api.NextcloudRetrofitServiceMethod.bodyToStream(NextcloudRetrofitServiceMethod.java:221)
	... 26 more
Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
	at libcore.io.Linux.open(Native Method)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
	at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:8046)
	at libcore.io.IoBridge.open(IoBridge.java:482)
	... 32 more

```

**Comment by stefan-niedermann** [comment]
Looks like zhis image `/2021-07-28_BackGroundImage.jpg` could not be uploaded succesfully. Do you remember to which card you attached it? I guess it should be attached on the Android app but not in the web, is this correct?

**Comment by neerro** [comment]
Oh, yeah I suggest I know what the image is. I've tested possibility to upload an image within I haven't an access to instance (and yes, in mobile app). Just tried how it is workin in that case, and - no way, it doesn't work.

So, what I shal to do further?

**Comment by stefan-niedermann** [comment]
Uhm... Can you long press on this image in the Android app and then press <kbd>Delete</kbd>? ([I don't think so](https://github.com/nextcloud/deck/issues/2904), but please try it).

In the worst case you will need to remove your account from the Deck Android App and readd it, this should solve the problem. Make sure that everything else is synchronized though before doing this.

Alternatively deleting the card in the Android app should also solve the issue, but make sure to copy the content of it before doing this.

@desperateCoder we should catch cases where the file to upload is not available anymore and remove the entry from the (local) database, so it doesn't attempt it on every sync, let's talk about it on the next coding session.

**Comment by neerro** [comment]
> 
> 
> Uhm... Can you long press on this image in the Android app and then press Delete? ([I don't think so](https://github.com/nextcloud/deck/issues/2904), but please try it).
>
So, after this operation It looks like working. I have no see the error anymore. I have deleted the file as per your suggestion.
And one more thing, I couldn't open those file as It seems that It didn't upload completely, or there are some bugs in uploaded process.
>

**Comment by neerro** [comment]
For additional information...
After all mentioned actions above, I've tried to upload an image once again (instance is available). When I choose file from mobile and press upload, I didn't see the file in the deck card then I've tried to repeat this action in second time but met the same situation then I've tried in third time after that opened the error window. I closed the error window (error code below) and not until then I could see the uploaded image.

```
App Version: 1.19.3
App Version Code: 1019003
Server App Version: 1.4.3
App Flavor: play

Files App Version Code: 30160190

---

OS Version: 4.14.116(10.0.0.221C10)
OS API Level: 29
Device: HWHRY-H
Manufacturer: HUAWEI
Model (and Product): HRY-LX1 (HRY-LX1RU)

---

java.io.IOException: Failed to create cacheFile
    at it.niedermann.nextcloud.deck.util.FilesUtil.getTempCacheFile(FilesUtil.java:73)
    at it.niedermann.nextcloud.deck.util.FilesUtil.copyContentUriToTempFile(FilesUtil.java:36)
    at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.lambda$uploadNewAttachmentFromUri$11$CardAttachmentsFragment(CardAttachmentsFragment.java:410)
    at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment$$ExternalSyntheticLambda1.run(Unknown Source:6)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:929)

```

**Comment by neerro** [comment]
In last case, when image was synch with the instance I have no capability to delete image (by long press on this image in the Android app and then press ```Delete```) anymore from mobile app and server deck as well.

**Comment by stefan-niedermann** [comment]
Allright, we think we found the reason for this problem and we fixed it in the future `1.19.4` release. Thanks for the detailed bug report and let us know in case the issue occurs again with `1.19.4` or later :slightly_smiling_face:.
